posting/list.go

```diff
@@ -1687,16 +1685,62 @@ func (l *List) Uids(opt ListOptions) (*pb.List, error) {
 		return out, nil
 	}
 
+	absFirst := opt.First
+	if opt.First < 0 {
+		absFirst = -opt.First
+	}
+	preAllowcateLength := min(absFirst, l.mutationMap.len()+codec.ApproxLen(l.plist.Pack))
+	if opt.Intersect != nil {
+		preAllowcateLength = min(preAllowcateLength, len(opt.Intersect.Uids))
+	}
+	// Pre-assign length to make it faster.
+	res := make([]uint64, 0, preAllowcateLength)
+
+	checkLimit := func() bool {
+		// We need the last N.
+		// TODO: This could be optimized by only considering some of the last UidBlocks.
+		if opt.First < 0 {
+			if len(res) > -opt.First {
+				res = res[1:]
+			}
+		} else if len(res) > opt.First {
+			return true
+		}
+		return false
+	}
+
+	if opt.Intersect != nil && len(opt.Intersect.Uids) < l.mutationMap.len()+codec.ApproxLen(l.plist.Pack) {
```

Comment:
same: use exact length. 
lets create another function inside list that intersects with the list of uids.